docs/HowToDevelopLoadableApps.md

```diff
@@ -0,0 +1,227 @@
+
+```
+CONFIG_EXAMPLES_MPU_TEST=y
+CONFIG_MPUTEST_KERNEL_CODE_ADDR=0x60000000
+CONFIG_MPUTEST_KERNEL_DATA_ADDR=0x20200000
+CONFIG_MPUTEST_APP_ADDR=0x80200100
+```
+   In above scenario, the configs description is as below:
+
+``` 
+CONFIG_MPUTEST_KERNEL_CODE_ADDR		This is same as FLASH address		CONFIG_FLASH_START_ADDR
+CONFIG_MPUTEST_KERNEL_DATA_ADDR		This is same as SRAM address		CONFIG_SRAM_START_ADDR
+CONFIG_MPUTEST_APP_ADDR			Application sections' address		CONFIG_RAM_REGIONx_START
+					(described below)
+```
+
+2. Other Loadable Apps test configs that could be enabled are as below:
+
+```
+#
+# Enable Test Scenarios
+#
+CONFIG_EXAMPLES_MESSAGING_TEST=y
+CONFIG_MESSAGING_TEST_REPETITION_NUM=1
+CONFIG_EXAMPLES_RECOVERY_TEST=y
+# CONFIG_ENABLE_RECOVERY_AGING_TEST is not set
+# CONFIG_EXAMPLES_BINARY_UPDATE_TEST is not set
+# CONFIG_EXAMPLES_MICOM_TIMER_TEST is not set
+CONFIG_EXAMPLES_ELF_FULLYLINKED=y
+```
+
+## Memory Requirements
+
+ELF is stored in flash partition and loaded into ram at runtime and hence we need to reserve
+both RAM & FLASH memories.
+
+### Flash Binary Partitions
+
+- Add a new flash partition for each application.
+- The 'size' should be large enough to accomodate the binary size (CONFIG_FLASH_PART_SIZE).
+- The 'type' should be of the type "bin" (CONFIG_FLASH_PART_TYPE).
+- The 'name' can be the name of the application (CONFIG_FLASH_PART_NAME).
+
+  In below scenario, the apps namely micom & wifi of sizes 1MB each are added. The two instances (wifi,wifi) accomodate FOTA update of binaries (if supported). One will contain the actual binary, whereas the other, the updated binary.
+
+```
+CONFIG_FLASH_PART_SIZE="2048,512,1024,1024,1024,1024,1024,128,128,"
+CONFIG_FLASH_PART_TYPE="kernel,none,bin,bin,bin,bin,smartfs,ftl,config,"
+CONFIG_FLASH_PART_NAME="kernel,app,micom,micom,wifi,wifi,userfs,ftl,config,"
+```
+
+### Loading Applications into RAM
+
+- Sections of loadable apps will be allocated from user RAM region.
+- So find out the memory region large enough to load the appliacations.
+- Add its address & size to CONFIG_RAM_REGIONx_START & CONFIG_RAM_REGIONx_SIZE (if not already present).
+- While loading applications, memory manager (MM) uses this address space to
+  allocate application's text & data sections.
+
+  In below scenario, this region starts at 0x80200000 of size 30MB.
+
+```
+CONFIG_RAM_REGIONx_START="0x20200000,0x80200000"
+CONFIG_RAM_REGIONx_SIZE="262144,31457280"
+```
+
+## Add new loadable apps configuration
+
+## Configuration Requirements
+
+- Follow the requirements as stated here ([Configuration Settings](#configuration-settings)) to enable all configurations related to loadable apps.
+
+## Memory Requirements
+
+- Follow the requirements as stated here ([Memory Requirements](#memory-requirements)) for reserving flash & ram partitions for adding new applications.
+
+## Code Changes Required
+
+- Add required apps to "os/loadable_apps/loadable_sample/<app_name>"
```

Comment:
New apps can be added to loadable_apps folder directly. loadable_sample folder is meant for sample apps.
Also, these folders are not inside os folder. Pls correct the path.